### ISSUE TITLE
try-catch-finally-else

### ISSUE THREAD

**Issue by akvadrako** [csvoss/onelinerizer#2]
Is it allowed to use anything from the standard library in generated code?

If so you can handle exceptions using the unit testing module like so:

```python
from __future__ import print_function
import sys, unittest

# input parameters
#tryme = lambda: (_ for _ in ()).throw(ValueError('ohno'))
#tryme = lambda: 'blah'
tryme = lambda: (_ for _ in ()).throw(AttributeError('ohno'))
exc = AttributeError
catch = lambda e: (print('failed (%r)' % e), 'caught')[1]
else_expr = lambda: print('else-expr')
finally_expr = lambda: print('finally')

# implementation
_catch = lambda e: state.__setitem__('result', catch(e[1]))
_throw = lambda e: (_ for _ in ()).throw(e[1])
result = type('tmp', (unittest.TestResult, ), {
    'addError': lambda s,t,e: _catch(e) if isinstance((e and e[1]), exc) else _throw(e)
})()
state = dict(r=result)
case = type('tmp', (unittest.TestCase, ), {
    'testMe': lambda s: (state.__setitem__('result', tryme()), else_expr())
})('testMe')
out = (case.run(state['r']), finally_expr, state['result'])[2]
```

**Comment by joshuamorton** [comment]
I'd consider this against the "spirit" of the idea, if you solve it in this manner, you open the door to doing things like

    import magic_file  #one line but shh
    magic_file.do_my_complex_operation()

which is "cheating", especially if we're saying that semicolons aren't in the spirit of the challenge.

You can do singly nested error handling with context managers, but its not a really extensible tool

**Comment by akvadrako** [comment]
Well, magic_file isn't in the standard library, so that's a step further than I've gone.

**Comment by joshuamorton** [comment]
right, but in this case all that unittest is doing is writing functions that wrap try catch finally.  We could do that ourselves, and it would be cheating.

**Comment by csvoss** [comment]
Personally, I find the standard library acceptable to use. :) It's mostly a matter of aesthetic taste.

**Comment by andersk** [comment]
Fixed by #56, #57, #58.
